index.js

```diff
@@ -0,0 +1,156 @@
+const exec = require('@actions/exec');
+const core = require('@actions/core');
+const github = require('@actions/github');
+const { parse: gitDiffParser } = require('what-the-diff');
+const fs = require('fs-extra');
+const simpleGit = require('simple-git');
+const env = process.env;
+
+function matchExact(r, str) {
+  let match = str.match(r);
+  return match && str === match[0];
+}
+
+function fixNewLineEOF(b) {
+  // Replace all trailing whitespaces
+  b = b.replace(/[ \t\n]*$/, '\n');
+
+  if (b.length === 1) {
+    // if the remaining character is a whitespace
+    if (/[ \t\n]*$/.test(b)) {
+      return '';
+    } else {
+      return b + '\n';
+    }
+  }
+  return b;
+}
+
+async function getChangedFilesPaths(pull_request, octokit, owner, repo) {
+  const { data: pullRequestDiff } = await octokit.pulls.get({
+    owner,
+    repo,
+    pull_number: pull_request.number,
+    mediaType: {
+      format: 'diff'
+    }
+  });
+
+  const parsedDiff = gitDiffParser(pullRequestDiff);
+
+  const changedFilePaths = parsedDiff.map((e) => {
+    return e['newPath'].replace('b/', '');
+  });
+
```

Comment:
What if the action is run on master branch ? Will this return all the files?

Reply:
The original aim was to only catch changed files in a PR. This only reads changed files in the current PR and will not work if run on the master branch directly.

Reply:
Okay. Please mention this in the readme and mention what will happen if it runs on master.

Also based on your comment in below code, the else part can be changed to log that this will work only on PRs and exit the code
```
 if (github.context.eventName == 'pull_request') {
       branch = github.context.payload.pull_request.head.ref;
     } else {
       branch = github.context.ref.replace('refs/heads/', '');
     }
```

Reply:
Should I do a simple `core.error` to log the statement or a `core.setFailed` to exit with failed status? What is more preferable in this scenario?

Reply:
`core.error` based on your code structure,  It is better to keep only one exit point.

Reply:
Done.